nav2_system_tests/src/system/test_system_launch.py

```diff
@@ -40,15 +43,35 @@ def generate_launch_description():
     bringup_dir = get_package_share_directory('nav2_bringup')
     params_file = os.path.join(bringup_dir, 'params', 'nav2_params.yaml')
 
-    # Replace the `use_astar` setting on the params file
-    param_substitutions = {
-        'planner_server.ros__parameters.GridBased.use_astar': os.getenv('ASTAR')}
+    # Replace the default parameter values for testing special features
+    # without having multiple params_files inside the nav2 stack
+    context = LaunchContext()
+    param_substitutions = {}
+
+    if (os.getenv('ASTAR') == "True"):
+        param_substitutions.update({'use_astar': "True"})
+        print("ASTAR set True")
+
+    if (os.getenv('GROOT_MONITORING') == "True"):
+        param_substitutions.update({'enable_groot_monitoring': "True"})
+        print("GROOT_MONITORING set True")
+
+    if (os.getenv('GROOT_ZMQ_PUB_PORT')):
```

Comment:
Are you actually changing these ports in the test? If not, can you remove these 2 variables since the defaults are fine?

Reply:
sure, removed